### ISSUE TITLE
# Issue: `train.py` Script Fails Due to Unrecognized Option `--data`

### ISSUE THREAD

**Issue by MohammadParsaTheFirst** [tqch/score-forgetting-distillation#2]
# 🐛 Issue: `train.py` Script Fails Due to Unrecognized Option `--data`

## Description

Hi,  
Thank you for your great work on this project!

I encountered an issue while running the script via `run_sfd.sh`. Specifically, the script fails due to an unrecognized command-line option: `--data`.

Here’s the command I executed:

```bash
CUDA_VISIBLE_DEVICES=0,1,2,3 NUM_GPUS=4 BATCH_SIZE=8 BATCH_SIZE_PER_GPU=2 \
SG_LEARNING_RATE=0.000003 G_LEARNING_RATE=0.000001 \
FORGET_DATA_PROMPT_TEXT="prompts/brad_pitt_prompts.txt" \
FORGET_DATA_PROMPT_TEXT_VAL="prompts/brad_pitt_prompts_val.txt" \
CONCEPT_TO_FORGET="brad pitt" \
CONCEPT_TO_OVERRIDE="a middle aged man" \
bash run_sfd.sh
```

Upon execution, I received the following error:
```bash
Usage: train.py [OPTIONS]
...
Error: no such option: --data
```

To further investigate, I ran:
```bash
python train.py --help
```
And confirmed that `--data` is not a recognized argument. The available related options include:
```bash
--data_stat ZIP|DIR: Path to the dataset stats

--data_prompt_text ZIP|DIR: Path to training prompts [required]
```

There is no `--data` flag listed among the options. This mismatch suggests either:

A typo or outdated parameter in the bash script, or

A documentation mismatch between the `run_sfd.sh` script and the current version of train.py

Expected Behavior
The `run_sfd.sh` script should align with the expected arguments of train.py. If `--data` was deprecated or renamed, the script should be updated accordingly.

Suggested Fix
Please verify whether `--data` was intended to be: `--data_stat`  or `--data_prompt_text`

Or another argument entirely. It may also be because of another thing as well, the repo's read me file is suggesting to 
run `bash run_df2_stage2.sh` which is not in the `SD` Directory.

Updating the bash script to reflect the correct parameter should resolve the issue.

Additional Context
Let me know if I can provide any logs or additional info. I’d be happy to test any fix or PR.

Thanks again for your efforts and support!

Best regards,
M. Parsa Dini

**Comment by tqch** [comment]
Hello M. Parsa Dini,

Thank you for your interest in this codebase. In the previous commit, the inclusion of Stable Diffusion was more of a preview, as we had not yet fully adapted the code from the development phase for an official public release.

Upon checking the code, I can confirm the issue you mentioned with the SD experiments. I’ve included a quick fix in the recent commit 3771028a3f4906f13c64abdf6d12fe7ee456c509. Please let me know if you encounter any further issues.

Best regards,
Tianqi

**Comment by MohammadParsaTheFirst** [comment]
Thank you for your response — it was very helpful. I do have a question, though. In the shell script for Stable Diffusion, the `--data` parameter was set to the validation set of the `MS-COCO-256` dataset. However, since this wasn’t specified in detail, I used the `coco2017` validation dataset instead, using the following command:

```bash
wget http://images.cocodataset.org/zips/val2017.zip
```
Is this choice appropriate? Also, could you clarify which dataset your team used for validation?

Yours sincerely,
M. Parsa Dini

**Comment by tqch** [comment]
Hello there,

Thank you for asking. To clarify, the `--data` parameter is not used during training, as the proposed method is data-free. The `--data_prompt_text` parameter provides the path to the text prompts used for training. In this case, we use prompts from the `aesthetics_6_plus` dataset as the concepts to retain—i.e., $c_r$ in the paper.

Please let me know if you have any further questions.

Best regards,  
Tianqi

**Comment by MohammadParsaTheFirst** [comment]
Thanks
